docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,73 @@
+# Rosbag2 record and replay service
+
+## Context
+
+In [ros2/rosbag2#773](https://github.com/ros2/rosbag2/issues/773), it was recommended that record service requests feature should be added to rosbag2.  
+
+This design describe how to implement this feature based on current architecture and how to extend current command parameter.  
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service introspection topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic. 
+
+## How to distinguish service introspection topic in all recorded topics 
+
+Service introspection is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.  
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service introspection topic
+  
+  [NameSpace/]ServiceName/_service_event
+
+So if the string ends with `/_service_event` in topic name, it should be service introspection topic. Of course, it is possible that hidden topic has the same name. So message type is used for further check. The message type for service introspection must ends with "_Event" (e.g. example_interfaces/srv/AddTwoInts_Event). Therefore, service introspection topic can be identified.
+
+### Expand the 'record' command
+
+Add 1 parameter.
+
+- `-S [SerivceName1 ServiceName2 ...]` or `--services [SerivceName1 ServiceName2 ...]`
+  
+    If the user doesn't specify service names, all service introspection topics are recorded.
```

Comment:
as user experience, this could be opposite? i think most likely that user wants to record the topics but service. so having this option as `opt-in` would be better to avoid unnecessary topics to be recorded?

Reply:
Yes. 
Users can decide whether to set `-S` or `--services` or not.   
If user doesn't set `-S or --services`, rosbag doesn't record service (service event topic).

Reply:
I don't want to change previous behavior. So '--all' only means all topic (Not include special service event topic).  
To make it clearer, '--all' may be changed to '--all-topics'.   
Or like service parameter, '--all' and specified topics are combined to '--topics [TopicName ...]' (If no topic name is specified, all topics are recorded). But this change is a little big for user.

Reply:
@Barry-Xu-2018 As regards to the 
> I don't want to change previous behavior. So '--all' only means all topic (Not include special service event topic).

As far as I remember we agreed on the version 
https://github.com/ros2/rosbag2/pull/1359/files#diff-cc475e77d25d214937d96658f496944c9dbf1b0dd6e01fc776528672300841bcR62
```
| --all | Record all topics and service event topics. Other hidden topics are excluded. |
```
I am closing this thread as resolved. If the proposed resolution is not complete or not as expected please feel free to mark it as unresolved again.

Reply:
> As far as I remember we agreed on the version

Yes. https://github.com/ros2/rosbag2/pull/1359#discussion_r1211203374 is old comments.   
We have reached a consensus on this.